# 异常

> 原文:[https://www.learnjavaonline.org/en/Exceptions](https://www.learnjavaonline.org/en/Exceptions)

* * *

每次发生错误时都会引发异常。可以从 http://docs . Oracle . com/javase/7/docs/API/Java/lang/exception . html 访问所有内置异常的列表。

使用 try/catch 语句处理异常。所有可能引发异常的代码都必须遵循 Catch 或 Specify 要求。要遵循这一要求，只需将可能引发错误的代码包装在 try 块中。如果由于某种原因，它不合适或者您不能使用 try/catch，您必须使用`throws`关键字指定一个方法/函数可以抛出的所有异常

```
public void writeFile() throws IOException 
```

您也可以使用 throw new 在代码中引发异常:

```
throw new IllegalArgumentException("Number not above 0");
/* Will print 
    Exception in thread "Main": java.lang.IllegalArgumentException: Number not above 0
*/ 
```

使用 try/catch 处理异常，这将在前面的课程中介绍:

```
try {
    System.out.println(arr[10]);
catch (ArrayIndexOutOfBoundsException ex) {
    System.out.println("Error in try block");
} 
```

## 锻炼

在教程代码中编写代码，以便在(n < 0)时抛出 IllegalArgumentException。它应该显示“错误”作为描述。提示:查看预期的输出，以及第二段代码。